app/controllers/admin/api/access_tokens_controller.rb

```diff
@@ -35,8 +35,52 @@ def create
     respond_with access_token
   end
 
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/users/{user_id}/access_tokens/{id}.json"
```

Comment:
Well we should also be able I think to read an access token from its `value` not only with its `id`

WDYT @3scale/system 

It would really be impractical for an administrator to use the id instead of the value.

Reply:
Also the user_id is not necessary as it should be the same user.
We are not allowing a user to change another's token

Reply:
I like this:
> Well we should also be able I think to read an access token from its value not only with its id

Not so sure about this though:
> Also the user_id is not necessary as it should be the same user.

Because it would be super weird to have the create (already in production) with one url and the show and delete with another one 🤔

Reply:
Current situation: We have in production right now a `create` with a url with `user_id` that can be used with `access_token` of user and with `provider_key` of `account`. The user can only use it for itself (otherwise it is unauthorized) and the account can for any of its users.

Reply:
with `provider_key` user would be able to create an access token for another user, which is not desired.. so we wether drop support for `provider_key` in this case or add `user_id` to all endpoints to mirror `create` method.. discussion in progress.

Reply:
@hallelujah @guicassolato @thomasmaas @macejmic 
So what should we do about this?

Reply:
The system/ruby team had a meeting about this and I will explain here what we decided. We informed product about this and I will also update the description in [THREESCALE-1766](https://issues.jboss.org/browse/THREESCALE-1766)

### What we have right now
We have only 1 API endpoint for access tokens but we have much more in the UI. The API endpoint that we have right now was created years ago before we were open source in https://github.com/3scale/system/pull/8441 but it was not documented and officially not supported and we documented it and made it public recently in https://github.com/3scale/porta/pull/430
This endpoint right now is under the URL `provider-url/admin/api/users/:user_id/access_tokens`. An officially documented that customers can authenticate through access token and we will allow to create only if the user_id belongs to the same user. Which is true but we also have (not documented and officially unsupported) the possibility to authenticate through provider_key and then we will allow to create for any user of that same account.
We also need to keep in mind that apicast may need to authenticate through provider_key 😄 

### What we we aim to have
We want it to be similar to what we have in the UI. So what we will do the next endpoints.

#### Index/Search Access Token API endpoint
Under the URL `provider-url/admin/api/access_tokens`. It will authenticate through the access_token and will return all the access tokens of the same user of the access_token if no extra parameter is sent. We will also allow the user to send a `name` string query param (like `provider-url/admin/api/access_tokens?name=example`) and we will return all the access tokens of that current user which name contains that string of the param.
For each access token we will return the following attributes.
  - scopes
  - permission
  - name
We will **not** return their values for security reasons 😄 

#### Show Access Token API endpoint
Under the URL `provider-url/admin/api/access_tokens/ID` or `provider-url/admin/api/access_tokens/VALUE`
It will authenticate through the access_token and only that same user can fetch an access token of its own. It can be fetched by `ID` or by its `value`.
We will return for the access token the same parameters as for the index (the previous endpoint).

#### Delete Access Token API endpoint
Under the URL `provider-url/admin/api/access_tokens/ID` or `provider-url/admin/api/access_tokens/VALUE`
It will authenticate through the access_token and only that same user can delete an access token of its own. It can be fetched by `ID` or by its `value`.

#### Create Access Token API endpoint
We will have a new endpoint that does exactly the same as we have, but with a different URL, under `provider-url/admin/api/access_tokens`, and this new endpoint will allow to authenticate only through access_token and not through provider_key.
The endpoint that we had until now will keep working but will be deprecated.

Reply:
An idea for the new endpoints could be namespacing them under a keyword to identify it's personal-relation stuff, such as `personal` or `new`. For example:
- `provider-url/admin/api/my/access_tokens`

Eventually, for notifications preferences:
- `provider-url/admin/api/my/notification_preferences`

I understand would means moving controllers and tests, but in case we think it's worth it, better now than having many endpoints to deprecate in the future.

Reply:
Ok, I will check and let you know about it @guicassolato 👍

Reply:
I like the idea, would probably call it `/api/personal/…` instead of `me` but if it implies too much work might not be worth it

Reply:
I think like Thomas, I like it but if this has to be done relatively quickly... that may be too much work 🤔 Although maybe not, I dunno, I will try 🤔